packages/react/src/components/Flag/Flag.tsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {Typography} from '@mui/material';
+import {FC, ReactElement} from 'react';
+import WorldFlag from 'react-world-flags';
+import {WithWrapperProps} from '../../models';
+import {composeComponentDisplayName} from '../../utils';
+
+export interface FlagsProps extends React.HTMLAttributes<HTMLElement & SVGElement> {
+  /**
+   * The two-letter/three-letter/three-digit country code of the flag.
+   */
+  countryCode: string;
+  /**
+   * The height of the flag.
+   */
+  height?: string;
+}
+
+const COMPONENT_NAME: string = 'Flag';
+
+const Flag: FC<FlagsProps> & WithWrapperProps = (props: FlagsProps): ReactElement => {
```

Comment:
WDYT about making this component name into something specific like `CountryFlag` because in the future we might need to have a flag icon in our icon library and it would be inconvenient to rename when importing or locating the component upon searching. 

or we can introduce a generic flag icon as a fallback when the countryCode doesnt match? 

E.g. - https://mui.com/material-ui/material-icons/?query=flag&selected=Flag